### PR TITLE
Comparison between accuracies of sklearn RFClassifier and ETClassifier under default parameter conditions

### LINKED ISSUE #27: Comparison between sklearn RandomForest and ExtraTrees
**Description**
Currently, there is no official comparison using multiple datasets between the accuracies of sklearn's RandomForest and ExtraTrees classifiers.

**Enhancement**
I will make a tutorial notebook that benchmark RFClassifier and ExtraTreesClassifier using the CC-18 and 100 datasets in OpenML. The figures that will be included are heatmaps (portraying accuracies of hyperparameter pairs), accuracies (plotted against number of features and number of samples), and runtimes (also plotted against number of features and number of samples).

### PR DESCRIPTION
**Pull request by sju777**
**Reference Issues/PRs**

- Fixes #27 in terms of comparing both classifiers under default parameter conditions

**What does this implement/fix? Explain your changes.**

- Gather accuracy and runtime data in benchmarking for sklearn RF and ET classifiers using the OpenML CC-18 and 100 suites
- Generate figures to compare accuracy and runtime data between RF and ET under default parameter conditions

**Any other comments?**

- none

PR is in progress, but any and all feedback is welcome!

### CODE REVIEW

**Comment by bdpedigo** [comment]
Notebook comments: 
- cell that says "Build the scikit-learn ExtraTrees classifier" is an RF classifier?
- May not want to expose your API key online (openML) 
- Seems like the 4 code blocks that are just for loops at the beginning could be functions that take in a classifier and a dataset suite to avoid redundant code
- same comment as the above for when you read in accuracies
- Figures in output cell [9]: I think you should make it more clear what is being plotted by labeling the x-axis as opposed to the `suptitle`. I think this would also be more clear with a dashed line at 0, and text on either side of the line saying "RF better", "ET better" as appropriate. If you have them stacked on top of each other I would have the two plots share an x axis. Finally, I wonder if in addition to the histogram, a [rugplot](https://seaborn.pydata.org/generated/seaborn.rugplot.html) would be informative, on the same axes. See the second two examples [here](https://seaborn.pydata.org/tutorial/distributions.html#plotting-univariate-distributions).
- For figures output from cell [11], make the title above the marginal plot.
- For figure output from cell [12], I don't think you need "diff </> 0 " text.
- Figure [15] same comment about title above text. also for all of these figures I think the title should just be "CC18/ OPenML100" you don't really need "accuracy difference vs features" because that should be clear from your axes labels. 
- [16] same comment about (diff < 0)
- [17] again, make these functions for readability
- [18] same comment about rugplot, just consider it and try it out, hard to say without seeing it. also same comment about the title, title should just be the dataset name, and then x-axis should definitely have label. probably y axis also
- [20] same comment about titles. yaxis label (difference) does not make it clear which is better. All of these plots having the "et better/rf better" text on either side of the line would probably be good.
- [22] same comments as the above

**Comment by bdpedigo** [comment]
remove `-checkpoint` file from PR. I accidentally reviewed that because it is the first thing i saw in `files changed` but they look to be the same? Let me know if any of my comments don't apply because there is a difference between the two.

**Comment by bdpedigo** [comment]
@sju777 nice work! I think overall things look good. Mainly, my comments fall under either small tweaks to make figures more clear, or minor code style (putting things into functions when they are repeated many times) to make the notebook shorter and the code more readable. But I don't think any of those comments should be a big deal to address/try out, overall the notebook is clear!

**Review by bdpedigo — DISMISSED** [review]
See comments above
